lib/zlib.js

```diff
@@ -170,95 +192,84 @@ function Zlib(opts, mode) {
   var strategy = Z_DEFAULT_STRATEGY;
   var dictionary;
 
-  if (typeof mode !== 'number')
-    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'mode', 'number');
-  if (mode < DEFLATE || mode > UNZIP)
-    throw new errors.RangeError('ERR_OUT_OF_RANGE', 'mode');
+  // The Zlib class is not exported to user land, the mode should only be
+  // passed in by us.
+  assert(typeof mode === 'number');
+  assert(mode >= DEFLATE && mode <= UNZIP);
 
   if (opts) {
     chunkSize = opts.chunkSize;
-    if (chunkSize !== undefined && chunkSize === chunkSize) {
-      if (chunkSize < Z_MIN_CHUNK || !Number.isFinite(chunkSize))
-        throw new errors.RangeError('ERR_INVALID_OPT_VALUE',
-                                    'chunkSize',
-                                    chunkSize);
-    } else {
+    if (!validateFiniteNumber(chunkSize, 'options.chunkSize')) {
       chunkSize = Z_DEFAULT_CHUNK;
+    } else if (chunkSize < Z_MIN_CHUNK) {
+      throw new errors.RangeError('ERR_OUT_OF_RANGE', 'options.chunkSize',
+                                  `>= ${Z_MIN_CHUNK}`, chunkSize);
     }
 
     flush = opts.flush;
-    if (flush !== undefined && flush === flush) {
-      if (flush < Z_NO_FLUSH || flush > Z_BLOCK || !Number.isFinite(flush))
-        throw new errors.RangeError('ERR_INVALID_OPT_VALUE', 'flush', flush);
-    } else {
+    if (!validateFiniteNumber(flush, 'options.flush')) {
       flush = Z_NO_FLUSH;
+    } else if (flush < Z_NO_FLUSH || flush > Z_BLOCK) {
+      throw new errors.RangeError('ERR_OUT_OF_RANGE', 'options.flush',
+                                  `>= ${Z_NO_FLUSH} and <= ${Z_BLOCK}`,
+                                  flush);
```

Comment:
Suggestion: all cases that check for both bounds (that are all but one if I checked correct) could also be moved in a separate functions.

That way it would look like e.g.:

```js
if (!validateFiniteNumber(flush, 'options.flush')) {
  arg = foo;
} else {
  validateRange(flush, Z_NO_FLUSH, Z_BLOCK, 'options.flush')
}
```

Or you actually combine the check and move the range check into the `validateFiniteNumber`:

```js
if (!validateNumberRange(flush, Z_NO_FLUSH, Z_BLOCK, 'options.flush'))
  flush = Z_NO_FLUSH;
```